### PR TITLE
added code for task 10

### PR DESCRIPTION
**Pull request by UsmanAJabbar**

### CODE REVIEW

File: api/v1/views/users.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/python3
+"""Cities"""
```
Comment (MenacingManatee) [inline api/v1/views/users.py]: This docstring should probably be 'Users' not 'Cities'

File: api/v1/views/users.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    from models.user import User
+    method = request.method
+
+    if method == 'GET':
+        if user_id is None:
+            users = storage.all(User).values
+            return jsonify([user.to_dict() for user in users])
+        else:
+            obj = storage.get(User, user_id)
+            return abort(404) if obj is None else jsonify(obj.to_dict())
```
Comment (MenacingManatee) [inline api/v1/views/users.py:20]: I like the way this was streamlined

File: api/v1/views/users.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    elif method == 'DELETE':
+        if user_id is None or storage.get(User, user_id) is None:
+            abort(404)
+        else:
+            storage.delete(User, user_id)
```
Comment (MenacingManatee) [inline api/v1/views/users.py]: Delete takes an object, not a class and an id. You need to change this to send it an object (probably using the storage.get method)

File: api/v1/views/users.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        if user_id is None or user is None:
+            abort(404)
+        if request.is_json:
+            json = request.get_json()
+            for key, value in json.items():
+                setattr(user, key, val)
+            user.save()
+            return jsonify(user.to_dict())
+        else:
+            abort(404, 'Not a JSON')
```
Comment (MenacingManatee) [inline api/v1/views/users.py]: This abort should be a 400 error, not 404

**Review by MenacingManatee — CHANGES_REQUESTED** [review]

**Review by UsmanAJabbar — COMMENTED** [review]
Updated the files.

**Pull request merged**